tests/end_to_end/test_runtimes_e2e.py

```diff
@@ -1,10 +1,10 @@
-from distutils.dir_util import copy_tree
+# Use shutil.copytree directly instead of distutils.dir_util.copy_tree
+# This avoids the type incompatibility issue
```

Comment:
This comment probably doesn't need to be here (you can add it to the commit message if you want). It talks about the change you're making, not about the existing code. It doesn't make sense to have a comment about `distutils`, when it's not in the code anymore.

Also, we already use `shutil.copytree` in other parts of the code.